### PR TITLE
vscode-extensions.WakaTime.vscode-wakatime: init at 1.2.3

### PR DESCRIPTION
**Pull request by eadwu**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/vscode-extensions/wakatime/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      sha256 = "1cf4a4a3e0c35f293124e5613c29cfec850f67c0c43cec2ce1d8cc2e83aa217f";
+    };
+
+    postPatch = ''
+      mkdir -p out/wakatime-master
+
+      cp -rt out/wakatime-master --no-preserve=all ${commandLineInterface}/*
+    '';
+
+    meta = with stdenv.lib; {
```
Comment (pSub) [inline pkgs/misc/vscode-extensions/wakatime/default.nix]: Please provide a description.

File: pkgs/misc/vscode-extensions/wakatime/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchgit, vscode-utils }:
+
+let
+  inherit (vscode-utils) buildVscodeMarketplaceExtension;
+
+  commandLineInterface = stdenv.mkDerivation {
+    name = "wakatime";
```
Comment (pSub) [inline pkgs/misc/vscode-extensions/wakatime/default.nix]: It is mandatory that the version is included in the name.

**Review by pSub — CHANGES_REQUESTED** [review]

File: pkgs/misc/vscode-extensions/wakatime/default.nix (hunk 1 above)
Comment (eadwu) [reply pkgs/misc/vscode-extensions/wakatime/default.nix]: Did it, just below

File: pkgs/misc/vscode-extensions/wakatime/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchgit, vscode-utils }:
+
+let
+  inherit (vscode-utils) buildVscodeMarketplaceExtension;
+
+  cliVer = "10.2.1";
+  commandLineInterface = stdenv.mkDerivation {
+    name = "wakatime-${cliVer}";
+
+    src = fetchgit {
```
Comment (Mic92) [inline pkgs/misc/vscode-extensions/wakatime/default.nix]: Unless this repository uses git submodules, please use `fetchFromGitHub`.

File: pkgs/misc/vscode-extensions/wakatime/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchgit, vscode-utils }:
+
+let
+  inherit (vscode-utils) buildVscodeMarketplaceExtension;
+
+  cliVer = "10.2.1";
+  commandLineInterface = stdenv.mkDerivation {
```
Comment (Mic92) [inline pkgs/misc/vscode-extensions/wakatime/default.nix]: We have already a wakatime package, can you reuse it here?

Comment (eadwu) [reply pkgs/misc/vscode-extensions/wakatime/default.nix]: It seems to directly call `cli.py` from the it's own internal download. The same files seem to located in `"${wakatime}/lib/python3.6/site-packages/wakatime"` so I'll try that. Should I copy over the `__pycache__` as well?
```
  ERR EROFS: read-only file system, open '/nix/store/wgic0b48qr7s0gl1540j6ivqp5z92i06-vscode-extension-WakaTime-vscode-wakatime-1.2.2/share/vscode/extensions/WakaTime.vscode-wakatime/out/wakatime-master.zip': Error: EROFS: read-only file system, open '/nix/store/wgic0b48qr7s0gl1540j6ivqp5z92i06-vscode-extension-WakaTime-vscode-wakatime-1.2.2/share/vscode/extensions/WakaTime.vscode-wakatime/out/wakatime-master.zip'
```
```
/nix/store/rsa1rsd6c6szy0dmi61599b33aqiy269-python3-3.6.5/bin/python3: can't open file '/nix/store/wgic0b48qr7s0gl1540j6ivqp5z92i06-vscode-extension-WakaTime-vscode-wakatime-1.2.2/share/vscode/extensions/WakaTime.vscode-wakatime/out/wakatime-master/wakatime/cli.py': [Errno 2] No such file or directory
```

Comment (eadwu) [reply pkgs/misc/vscode-extensions/wakatime/default.nix]: Works with current wakatime version though it still attempts to update. Should be fixed with #42238.

**Pull request merged**